lib/rdf/repository.rb

```diff
@@ -285,14 +304,10 @@ def has_statement?(statement)
       # @see RDF::Enumerable#each_statement
       def each_statement(&block)
         if block_given?
-          # Note that to iterate in a more consistent fashion despite
-          # possible concurrent mutations to `@data`, we use `#dup` to make
-          # shallow copies of the nested hashes before beginning the
-          # iteration over their keys and values.
-          @data.dup.each do |g, ss|
-            ss.dup.each do |s, ps|
-              ps.dup.each do |p, os|
-                os.dup.each do |o|
+          @data.each do |g, ss|
+            ss.each do |s, ps|
+              ps.each do |p, os|
+                os.each do |o|
                   # FIXME: yield has better performance, but broken in MRI 2.2: See https://bugs.ruby-lang.org/issues/11451.
                   block.call(RDF::Statement.new(s, p, o, graph_name: g.equal?(DEFAULT_GRAPH) ? nil : g))
```

Comment:
This is patched in 2.2.4 (and in 2.3.0 and above). See: Mon Nov 23 00:19:51 2015  Koichi Sasada  <ko1@atdot.net>

Should we switch back to `yield` and declare `2.2` versions before `2.2.4` unsupported?